eras/shelley/impl/src/Cardano/Ledger/Shelley/Genesis.hs

```diff
@@ -207,7 +207,7 @@ data ShelleyGenesis c = ShelleyGenesis
   , sgMaxLovelaceSupply :: !Word64
   , sgProtocolParams :: !(PParams (ShelleyEra c))
   , sgGenDelegs :: !(Map (KeyHash 'Genesis c) (GenDelegPair c))
-  , sgInitialFunds :: LM.ListMap (Addr c) Coin
+  , sgInitialFunds :: !(LM.ListMap (Addr c) Coin)
```

Comment:
I'll add comments noting the intentional laziness in the `sgInitialFunds` and `sgStaking` fields -- would you be happy for me to add an explicit but redundant `~` annotation to both of those fields too, or is that overkill?

Reply:
I wish we could do that, unfortunately lazy annotation is an error without `StrictData`. Except we don't want to use `StrictData`, since it can lead to more unexpected bugs than being lazy by default. Here is what you'll get if you try do: `data Foo a = Foo ~a`

```haskell
error: …
    • Lazy annotation (~) without StrictData
```

Reply:
thanks for catching this @lehins , I wouldn't have realized the problem

Reply:
ah, of course! I'm very much used to having `StrictData` on all the time lol

shame that the discussions around making GHC allow `~` without `StrictData` never went anywhere